### DISCUSSION TITLE
showing content in an other section of the current component

### DISCUSSION THREAD

**Discussion by nielsnl68** [preactjs/preact#3492] [Q&A] [answered]
I'm new to preact and i'm still learning, so i hope to find some good places where to look for good information.

My webpage looks something like this:
```
<body>
   <header>
      <logo />
       <navigation-links />
       <action-buttons />
   </header>
   <page-section>
</body>
```

the header is only refreshed only when you navigate to an other page.
where the content of the `action-buttons` should be dynamic based on the content state of the page.
I tried already a couple of thinks myself but i failed so far.

**My question is how can i add content inside `action-buttons` element while rendering an the `page-section'?**

**Comment by JoviDeCroock** [comment, marked as answer]
Preact causes re-render when state changes so whatever the page-section is subscribes to can change the content of action-buttons/... if you subscribe to that state within that component through context/...

**Comment by nielsnl68** [comment]
Ha, now i understand what was happen. Thanks for pointing this issue out.
Could it be done using a component that is used on rendering the the `page-section` like:

```
<page-section>
    ... content ...
    <actionbuttons>
       <link ... >action 1</link>
       <link ... >action 2</link>
       <link ... >action 3</link>
     </actionbuttons>
</page-section>
```

where the actionbuttons component places the child elements under the `action-buttons` element of the example i shown above?
